docs/2_5DSPH.rst

```diff
@@ -0,0 +1,187 @@
+=================
+
+SPH method can be used to solve depth integrated equations where a 2D
+(respectively 3D) equation is reduced to a 1D (respectively 2D) one.
+This is used in ocean engineering to solve shallow water equations (SWE)
+in open or closed channels for example. In all these applications,
+whether it is 1D or 2D SPH, the fluid is most of the time,
+assumed to move on a horizontal plane (bed elevation is set to a constant).
+In the case of avalanche flow, the "bed" is sloped and irregular.
+The aim is to adapt the SPH method to apply it to depth integrated equations
+on a 2D surface living in a 3D world.
+
+Method
+------
+The SPH method is used to express a quantity (the flow depth in our case) and
+its gradient at a certain particle location as a weighted sum of its neighbors
+properties. The principle of the method is well described in :cite:`LiLi2010`.
+In the case a depth integrated equations (for example SWE), a scalar function
+:math:`f` and its gradient can be expressed as following:
+
+.. math::
+    f_{i} &= \sum\limits_{j}f_{j}A_{j}\,W_{ij}\\
+    \mathbf{\nabla}f_{i} &= -\sum\limits_{j}f_{j}A_{j}\,\mathbf{\nabla}W_{ij}
+    :label: sph formulation
+
+Which gives for the flow depth:
+
+.. math::
+    \overline{h}_{i} &= \frac{1}{\rho_0}\,\sum\limits_{j}{m_{j}}\,W_{ij}\\
+    \mathbf{\nabla}\overline{h}_{i} &= -\frac{1}{\rho_0}\,\sum\limits_{j}{m_{j}}\,\mathbf{\nabla}W_{ij}
+    :label: sph formulation for fd
+
+Where :math:`W` represents the SPH-Kernel function.
+
+The computation of its gradient depends on the coordinate system used.
+
+.. _standard-method:
+
+Standard method
+~~~~~~~~~~~~~~~~~
+
+Let us start with the computation of the gradient of a scalar function
+:math:`f \colon \mathbb{R}^2 \to \mathbb{R}` on a horizontal plane.
+Let :math:`P_i=\mathbf{x}_i=(x_{i,1},x_{i,2})` and :math:`Q_j=\mathbf{x}_j=(x_{j,1},x_{j,2})` be two points in :math:`\mathbb{R}^2` defined by
+their coordinates in the Cartesian coordinate system :math:`(P_i,\mathbf{e_1},\mathbf{e_2})`. :math:`\mathbf{r}_{ij}=\mathbf{x}_i-\mathbf{x}_j` is the vector going from
+:math:`Q_j` to :math:`P_i` and :math:`r_{ij} = \left\Vert \mathbf{r}_{ij}\right\Vert` the length of this vector.
+Now consider the kernel function :math:`W`:
+
+
+.. math::
+  \left.
+  \begin{aligned}
+  W \colon \mathbb{R}^2 \times \mathbb{R}^2 \times \mathbb{R} &\to \mathbb{R}\\
+  (P_i, Q_j, r_0) &\mapsto W(P_i, Q_j, r_0)
+  \end{aligned}
+  \right.\quad, r_0\in\mathbb{R} \mbox{ is the smoothing kernel length}
+
+In the case of the spiky kernel, :math:`W` reads (2D case):
+
+.. math::
+   W_{ij} = W(\mathbf{x_i},\mathbf{x_j},r_0) = W(\mathbf{x_i}-\mathbf{x_j},r_0) = W(\mathbf{r_{ij}},r_0) = \frac{10}{\pi r_0^5}\left\{
+   \begin{aligned}
+   & (r_0 - \left\Vert \mathbf{r_{ij}}\right\Vert)^3, \quad &0\leq \left\Vert \mathbf{r_{lj}}\right\Vert \leq  r_0\\
+   & 0 , & r_0 <\left\Vert \mathbf{r_{ij}}\right\Vert
+   \end{aligned}
+   \right.
+   :label: kernel function
+
```

Comment:
Please split this equation, it is getting too wide on the docs. 
![image](https://user-images.githubusercontent.com/6945681/105222302-9f233100-5b5a-11eb-85b5-4de1814a77a4.png)

Same goes for equations 9, 13 of the com1DFA Kernel theory